java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/Configuration.kt

```diff
@@ -121,22 +122,54 @@ fun retrieveAdminUrl(): String {
     } else adminAddress?.toUrlString(false).toString()
 }
 
+fun convertToAgentArguments(options: String): AgentArguments {
+    logger.debug { "agent options:$options" }
+    val agentParameters = options.asAgentParams()
+    val configPath = agentParameters["configPath"] ?: getenv(SYSTEM_CONFIG_PATH)?.toKString()
+    logger.debug { "configFile=$configPath, agent parameters:$agentParameters" }
+    val agentParams = if (!configPath.isNullOrEmpty()) {
+        val properties = readFile(configPath)
+        logger.debug { "properties file:$properties" }
+        properties.asAgentParams("\n", "#")
+    } else {
+        logger.warn { "Deprecated. You should use a config file instead of options. It will be removed in the next release" }
+        agentParameters
+    }
+    logger.debug { "result of agent parameters:$agentParams" }
+    return agentParams.parseAs()
+}
+
+fun validate(args: AgentArguments) {
+    AgentArgumentsValidator.validate(args)
+}
+
+private fun String?.asAgentParams(
+    lineDelimiter: String = ",",
+    filterPrefix: String = "",
+    mapDelimiter: String = "="
+): Map<String, String> {
+    if (this.isNullOrEmpty()) return emptyMap()
+    return try {
+        this.split(lineDelimiter)
+            .filter { it.isNotEmpty() && (filterPrefix.isEmpty() || !it.startsWith(filterPrefix)) }
+            .associate {
+                it.substringBefore(mapDelimiter) to it.substringAfter(mapDelimiter, "")
+            }
+    } catch (parseException: Exception) {
+        throw IllegalArgumentException("wrong agent parameters: $this")
+    }
+}
+
+private fun readFile(filePath: String): String {
+    val fileDescriptor = open(filePath, EROFS)
+    if (fileDescriptor == -1) throw IllegalArgumentException("Cannot open the config file with filePath='$filePath'")
+    val bytes = Input(fileDescriptor).readBytes()
```

Comment:
I suggest wrapping this one and the next line into `try ... catch` to catch read/decode error and re-throw them with the clear error message